File: BehaviorPattern/Strategy/Golang/Concretes/PayPalStrategy.go
package concretes

import (
	"fmt"
	iface "v/Interfaces"
)

// Concrete Strategy for Credit Card Payment
type PayPalStrategy struct {
	emailId string
	password string
}

// Constructor
func NewPayPalStrategy(emailId string, password string) iface.PaymentStrategy {
	return &PayPalStrategy{
		emailId: emailId,
		password: password,
	}
}

// Override pay
func (c *PayPalStrategy) Pay(amount int) {
	fmt.Println(amount, " paid with pay pal.")
}

File: BehaviorPattern/Strategy/Python/Concretes/PayPalStrategy.py

from Interfaces import PaymentStrategy

class PayPalStrategy(PaymentStrategy):
    """
        Class PayPalStrategy implementation PaymentStrategy
    """
    def __init__(self, *args, **kwargs):
        self.emailId = kwargs.get('emailId', None)
        self.password = kwargs.get('password', None)
        
        if args:
            lenArgs = len(args)
            if lenArgs > 0:
                self.emailId = args[0]
            if lenArgs > 1:
                self.password = args[1]
        super().__init__(args, kwargs)
        
    # Override Pay 
    def pay(self, amount):
        print("Paid with PayPal: ", amount)